worker-policy-shared/src/main/java/com/github/cafdataprocessing/worker/policy/shared/Document.java

```diff
@@ -131,4 +137,23 @@ public void setPolicyDataProcessingRecord(
             DocumentProcessingRecord policyDataProcessingRecord ) {
         this.policyDataProcessingRecord = policyDataProcessingRecord;
     }
+
+    @Override
+    public void removeSubdocument(String reference)
+    {
+        documents.stream().forEach(doc -> {
+            if (doc.getReference().equals(reference)) {
+                documents.remove(doc);
```

Comment:
Should you exit the forEach once you have removed the subdocument requested. unless you think 2 docs with same reference to be removed, and if so should it be removeSubdocuments.